src/api/checkCompatibility.ts

```diff
@@ -0,0 +1,148 @@
+import {stripIndent, oneLine} from 'common-tags'
+import {Modules} from '../fs/modulesController'
+import {PnpmError, PnpmErrorCode} from '../errorTypes'
+import semver = require('semver')
+
+export type ProjectCompatibilityOptions = {
```

Comment:
nit: `*Options` suffix is overly general and while it is used across the codebase already I think it can be dropped. Just a nitpick though.